src/components/patch/PatchContainer.tsx

```diff
@@ -48,9 +60,40 @@ export class PatchContainer extends React.Component<Props, State> {
   private loadPatches() {
     this.props.client.getPatches((err, resp, body) => {
       const versions = ConvertToPatches(JSON.stringify(resp.body)).VersionsMap
-      this.setState({ versions: versions });
+      this.setState({ 
+        versions: versions, 
+        visible: versions, 
+      });
     }, this.props.client.username);
   }
+
+  private search = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const query = event.currentTarget.value;
+    const filteredPatches = this.filterItems(query);
+    this.setState({
+      visible: filteredPatches,
+    });
+  }
+
+  private filterItems(query: string) {
+    const filtered = {};
+    if(query === "") {
+      return this.state.versions;
+    }
+    Object.keys(this.state.versions).map(versionId => {
+      const patch = this.state.versions[versionId];
+      let description = patch.Version.message;
+      if (description === undefined) {
```

Comment:
There shouldn't be a need to default a description here. But if there is, the right way would be to create a separate function for the default description rather than copying the code from the patch component

Reply:
It is necessary here since the description is what we're searching on. I moved this code in to a function.

Reply:
Isn't it already set by the patch itself?

Reply:
It wasn't before, but I just changed that (I thought initially that the change wouldn't be reflected in the parent PatchContainer, only the Patch). The code looks cleaner/shorter/all around better now.